src/storage/src/hummock/store/state_store.rs

```diff
@@ -152,136 +131,18 @@ impl HummockStorageCore {
-                LevelType::Overlapping | LevelType::Unspecified => {
-                    let sstable_infos = prune_ssts(
-                        level.table_infos.iter(),
-                        read_options.table_id,
-                        &(key..=key),
-                    );
-                    for sstable_info in sstable_infos {
-                        table_counts += 1;
-                        if let Some(v) = get_from_sstable_info(
-                            self.sstable_store.clone(),
-                            sstable_info,
-                            &internal_key,
-                            read_options.check_bloom_filter,
-                            &mut local_stats,
-                        )
-                        .await?
-                        {
-                            // todo add global stat to report
-                            local_stats.report(self.stats.as_ref());
-                            return Ok(v.into_user_value());
-                        }
-                    }
-                }
-                LevelType::Nonoverlapping => {
-                    let mut table_info_idx = level.table_infos.partition_point(|table| {
-                        let ord =
-                            user_key(&table.key_range.as_ref().unwrap().left).cmp(key.as_ref());
-                        ord == Ordering::Less || ord == Ordering::Equal
-                    });
-                    if table_info_idx == 0 {
-                        continue;
-                    }
-                    table_info_idx = table_info_idx.saturating_sub(1);
-                    let ord = user_key(
-                        &level.table_infos[table_info_idx]
-                            .key_range
-                            .as_ref()
-                            .unwrap()
-                            .right,
-                    )
-                    .cmp(key.as_ref());
-                    // the case that the key falls into the gap between two ssts
-                    if ord == Ordering::Less {
-                        continue;
-                    }
-
-                    table_counts += 1;
-                    if let Some(v) = get_from_sstable_info(
-                        self.sstable_store.clone(),
-                        &level.table_infos[table_info_idx],
-                        &internal_key,
-                        read_options.check_bloom_filter,
-                        &mut local_stats,
-                    )
-                    .await?
-                    {
-                        local_stats.report(self.stats.as_ref());
-                        return Ok(v.into_user_value());
-                    }
-                }
-            }
-        }
-
-        local_stats.report(self.stats.as_ref());
-        self.stats
-            .iter_merge_sstable_counts
-            .with_label_values(&["sub-iter"])
-            .observe(table_counts as f64);
+        let read_snapshot = read_filter_for_local(
```

Comment:
We use `read_filter_for_local` for HummockStorageCore while we use `read_filter_for_batch` for HummockStorage. This means that HummockStorageCore is used by local while HummockStorage is used by batch. Although this is by design and it is correct, I feel that it is too implicit so it can easily confuse others. Let's consider making it more explicit with #6154.

Reply:
Consistent with the above comments, the most critical question here is whether we support read umcommitted. I think this conclusion should be obtained first, and then simplified the local of read_filter (It means that Batch Query may not need to understand ReadVersion at all)